src/OrbitBase/ThreadUtilsLinux.cpp

```diff
@@ -22,12 +22,45 @@ namespace orbit_base {
 // including the terminating null byte ('\0')".
 static constexpr size_t kMaxThreadNameLength = 16;
 
-pid_t GetCurrentThreadId() {
+static constexpr int32_t kInvalidLinuxThreadId = -1;
+static constexpr int32_t kInvalidLinuxProcessId = -1;
+
+static constexpr uint32_t kIntMax = static_cast<uint32_t>(INT_MAX);
+
+uint32_t GetCurrentThreadId() { return GetThreadIdFromNative(GetCurrentThreadIdNative()); }
+
+uint32_t GetCurrentProcessId() { return GetProcessIdFromNative(GetCurrentProcessIdNative()); }
+
+bool IsValidThreadId(uint32_t tid) { return tid <= kIntMax && tid != kInvalidThreadId; }
+
+bool IsValidProcessId(uint32_t pid) { return pid <= kIntMax && pid != kInvalidProcessId; }
+
+pid_t GetCurrentThreadIdNative() {
   thread_local pid_t current_tid = syscall(__NR_gettid);
   return current_tid;
 }
 
-std::string GetThreadName(pid_t tid) {
+pid_t GetCurrentProcessIdNative() { return getpid(); }
+
+uint32_t GetThreadIdFromNative(pid_t tid) {
+  return tid >= 0 ? static_cast<uint32_t>(tid) : orbit_base::kInvalidThreadId;
+}
+
+uint32_t GetProcessIdFromNative(pid_t pid) {
+  return pid >= 0 ? static_cast<uint32_t>(pid) : orbit_base::kInvalidProcessId;
+}
+
+pid_t GetNativeThreadId(uint32_t tid) {
+  return IsValidThreadId(tid) ? static_cast<pid_t>(tid) : kInvalidLinuxThreadId;
+}
+
+pid_t GetNativeProcessId(uint32_t pid) {
```

Comment:
nit: ToNative?

Reply:
Yeah, I changed to "ToNativeX" and "FromNativeX", I think it better conveys the fact that it's a conversion function.